XXX: just a sample patch illustratin how to remove rosbuild
XXX: upstream should decide on a strategy: either plain cmake or ros-catkin

--- CMakeLists.txt.orig	2013-06-28 18:51:23.371807126 +0200
+++ CMakeLists.txt	2013-06-28 18:51:29.299785429 +0200
@@ -1,5 +1,4 @@
 cmake_minimum_required(VERSION 2.4.6)
-include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
 
 INCLUDE(cmake/base.cmake)
 INCLUDE(cmake/dist.cmake)
@@ -23,30 +22,26 @@
 
 # end of the partial extraction from SETUP_PROJECT
 
-
-#set(ROS_BUILD_TYPE RelWithDebInfo)
-set(ROS_BUILD_TYPE Debug)
-
-rosbuild_init()
-
 #set the default path for built executables to the "bin" directory
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 #set the default path for built libraries to the "lib" directory
 set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
 # Search for dependencies.
-ADD_REQUIRED_DEPENDENCY("jrl-dynamics")
-
-rosbuild_add_boost_directories()
-
-# Make sure Boost.Filesystem v2 is used.
-add_definitions(-DBOOST_FILESYSTEM_VERSION=2)
-
-rosbuild_add_library(${PROJECT_NAME} src/parser.cc)
-rosbuild_link_boost(${PROJECT_NAME} filesystem system)
-PKG_CONFIG_USE_DEPENDENCY(${PROJECT_NAME} jrl-dynamics)
+find_package(PkgConfig)
+pkg_check_modules(JRL jrl-dynamics jrl-mal)
+find_package(Boost 1.36.0 REQUIRED filesystem system)
+
+# targets 
+add_library(${PROJECT_NAME} SHARED src/parser.cc)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+include_directories(${Boost_INCLUDE_DIRS})
+include_directories(${JRL_INCLUDE_DIRS})
+link_directories(${JRL_LIBRARY_DIRS})
+target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${JRL_LIBRARIES})
 
-rosbuild_add_executable(display-robot examples/display-robot.cc)
-target_link_libraries(display-robot ${PROJECT_NAME})
+add_executable(display-robot examples/display-robot.cc)
+target_link_libraries(display-robot ${PROJECT_NAME} ${JRL_LIBRARIES})
 
-add_subdirectory(tests)
+# later...
+# add_subdirectory(tests)
